### PR TITLE
Organization of SSL following template

### PR DESCRIPTION
**Pull request by CassNot**
- Reorganization of the QSSL folder following the template
- Updated Readme accordingly and added results
- Added pretrained weights from SSL with classical, qiskit and MerLin representation head
- Added tests on InfoNCE loss and QSSL model to verify the sanity of the environment easily

### CODE REVIEW

File: qSSL/implementation.py [hunk 1]
```diff
@@ -140,10 +143,66 @@
...
+        )
+        args.activation = getattr(args, "activation", None) or cfg["model"].get(
+            "activation", "null"
+        )
+        args.shots = getattr(args, "shots", None) or cfg["model"].get("shots", 100)
+        # Merlin specific
+        args.modes = getattr(args, "modes", None) or cfg["model"].get("modes", 10)
+        args.no_bunching = getattr(args, "no_bunching", None) or cfg["model"].get(
+            "no_bunching", False
+        )
```
[the remark this answers is not in the document]
Comment (CassNot) [reply qSSL/implementation.py]: Edited in new commit

File: qSSL/README.md [hunk 2]
```diff
@@ -1,144 +1,143 @@
...
-- **`training_utils.py`**: Training utilities including InfoNCE loss and evaluation functions
+## Linear probing only
+Evaluate pretrained encoders with a frozen representation and train a linear head:
+```bash
+# Single checkpoint file
+python linear_probing.py --pretrained ./results/merlin/<timestamp>/model-cl-5-epoch-5.pth
 
-### Quantum Backends
+# Or point to a results directory (auto-discovers .pth files)
+python linear_probing.py --pretrained ./results/merlin/<timestamp>/
```
Comment (philippeschoeb) [inline qSSL/README.md:129]: It is not clear what happens when there are many model files (.pth) in a folder. Does it only consider the most pre-trained model ? Or does it load and run fine-tuning on every model one at a time ?

**Review by philippeschoeb — CHANGES_REQUESTED** [review]

**Comment by philippeschoeb** [comment]
Another thing, I am not sure whether the `__pycache__` folders are useful or not. If they are, that is okay but if not, they should be removed.

They are placed at several locations:
- `lib/__pycache__/`
- `lib/qnn/__pycache__/`
- `lib/qnn/activation_function/__pycache__/`
- `lib/qnn/ansatz/__pycache__/`
- `lib/qnn/input/__pycache__/`
- `tests/__pycache__`
